feat: Added example for openAI function call

RamK777-stack opened the pull request:
Added function call example in openAI.

/claim #469 
/closes #469

Note: 

I can't able to test the changes because I don't have an active openAI key. I consumed all of my free quota 🥺🥺

<img width="463" alt="image" src="https://github.com/triggerdotdev/trigger.dev/assets/69167444/44ea682e-5627-4cab-bf78-f8a335d6fa13">

ericallam commented on examples/job-catalog/src/openai.ts:
```diff
@@ -74,6 +127,38 @@ client.defineJob({
+      })
+      const finalCompletion = await createCompletion();
+      // @ts-ignore
+      await io.logger.info(finalCompletion?.data?.choices[0].message);
```
In general, could you add more comments in the code to clearly explain what is going on?

RamK777-stack replied:
Added more comments now !!

RamK777-stack commented on examples/job-catalog/src/openai.ts:
```diff
@@ -74,6 +127,38 @@ client.defineJob({
     });
+
+    //Function call example for OpenAI
+    const createCompletion = async () => {
```
In the first call, openAI returns JSON which contains the function name and arguments

Example:
{'index': 0,
 'message': {'role': 'assistant',
  'content': None,
  'function_call': {'name': 'get_n_day_weather_forecast',
   'arguments': '{\n  "location": "Glasgow, Scotland",\n  "format": "celsius",\n  "num_days": 5\n}'}},
 'finish_reason': 'function_call'}
 
 Using this we need to call our functions. and pass the result to GPT again for summarization. so we need 2 calls. 

Reference: https://github.com/openai/openai-cookbook/blob/main/examples/How_to_call_functions_with_chat_models.ipynb

RamK777-stack commented on examples/job-catalog/src/openai.ts:
```diff
@@ -74,6 +127,38 @@ client.defineJob({
+
+    //Function call example for OpenAI
+    const createCompletion = async () => {
+      const chatCompletion = await io.openai.createChatCompletion("chat-completion", {
```
I didn't understand this part @ericallam, 

Do I need to use await io.openai.chats.completions.create instead of await io.openai.createChatCompletion ?

RamK777-stack commented on examples/job-catalog/src/openai.ts:
```diff
@@ -10,6 +10,59 @@ export const client = new TriggerClient({
+  }
+]
+
+const get_current_affairs = async (query: string) => {
```
I have removed this now and sent mock response !!

RamK777-stack commented:
Thanks @ericallam 
1. I have added more comments.
2. Removed Google API request and mocked responses.
3. Fixed ts errors.

Please check and share your feedback.

RamK777-stack commented:
HI @ericallam Did you get a chance to look into this? I have made the changes.
